zephyr/lib/pm_runtime.c

```diff
@@ -74,6 +80,12 @@ void platform_pm_runtime_enable(uint32_t context, uint32_t index)
 		tr_dbg(&power_tr, "removing prevent on d0i3 (lock is active=%d)",
 		       pm_policy_state_lock_is_active(PM_STATE_RUNTIME_IDLE, PM_ALL_SUBSTATES));
 		break;
+#if CONFIG_ADSP_DYNAMIC_CLOCK_SWITCHING
```

Comment:
Btw, what's the long term plan to move this into Zephyr as I would expect we should all the platform PM logic in Zephyr.

Reply:
We won't be able to move everything related to PM to Zephyr. We need to prevent changes to the clock and dynamic power gating on the application side. What can definitely be moved to Zephyr is the custom power policy (several changes in this direction have already been made).

Reply:
ack, but we should do that with Zephyr PM APIs. e.g. ```zephyr_pm_prevent_clock_gating(CLOCK_ID)``` so that eventually we have zero IP code in SOF and its all in Zephyr.  I know we cant yet do this today, but just for the record that we may have to extend Zephyr APIs here.

Reply:
> zephyr_pm_prevent_clock_gating(CLOCK_ID)

btw, we have clock gating, this is about clock switching.

I was wondering if there is really a benefit from **idle** clock switching (with switching overhead) if we have hw clock **gating** anyway